src/nanosvg.h

```diff
@@ -2578,6 +2591,29 @@ static void nsvg__parseGradientStop(NSVGparser* p, const char** attr)
 	stop->offset = curAttr->stopOffset;
 }
 
+static void nsvg__parseGroup(NSVGparser* p, const char** attrs)
+{
+	NSVGgroup* group, *g;
+	NSVGattrib* attr = nsvg__getAttr(p);
+	
+	nsvg__parseAttribs(p, attrs);
+	if (!*attr->id) //skip anonymous groups
+		return;
+	
+	group = (NSVGgroup*)malloc(sizeof NSVGgroup);
+	memset(group, 0, sizeof NSVGgroup);
+	memcpy(group->id, attr->id, sizeof(group->id));
+	group->parent = attr->group;
+	attr->group = group;
+	
+	if (!p->image->groups) 
```

Comment:
Either use the same tail adding patterns as shapes used to, or use the optimized version (add front, reverse).